### ISSUE TITLE
ChangeLog (`ObjectChange`) - Add object snapshot activation feature (i.e "rollback")

### ISSUE THREAD

**Issue by vpogrebi** [nautobot/nautobot#2195]
### As ...

P.D. - Plugin Developer

### I want ...

Object "snapshot/version" activation ("rollback" so to speak) feature to be added to ChangeLog (`ObjectChange`) implementation 

### So that ...

Any object can be easily restored ("rolled back") to any particular past snapshot at any given moment

### I know this is done when...

When both ChangeLog table view (a list of object's change log records) and detail view (particular change log's detail view) have `Restore` (or `Rollback`) button, as well as an `ObjectChange` method (`restore()` or `rollback()`) is added - that restores object to that particular state.

## SPECIFIC DETAILS

At a high level, this request - is to add a "restore" (or "rollback") ChangeLog feature that would allow restore an object to any particular past state preserved in (represented by) an existing ObjectChange record. In other words - past object's "snapshot" (`object_data`) would be used to update model's object attributes. A button can be added to both ChangeLog's table view, and detail view - to allow restore from the UI, and a new method can be added that exposes same functionality through Python code.

Optionally (but desired), additional `version` (or `snapshot_version`) attribute can be added to `ObjectChange` model - that can be used to identify each ObjectChange record as an object's "version", and used as an argument passed to `restore()` (or `rollback()`) method.

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [X] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [X] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

Existing `extras.models.change_logging.ObjectChange` model should be updated:

1. New attribute (`version` or `snapshot_version`) added
2. New method (`restore()` or `rollback()`) added

**Comment by glennmatthews** [comment]
Something tricky to consider when thinking about this feature is foreign-keys. If an IPAddress "10.1.1.1/24" is associated with "Device A" interface "Eth0" at time X, but later reassigned to "Device B" interface "Eth1" at time Y, we may not be able to roll back the IPAddress to time X if Device A or Eth0 no longer exists in the present time.

**Comment by vpogrebi** [comment]
Yes, this can be a limitation to this feature's implementation (there may be more limitations like this - like underlying model changes and some fields no longer there). But... at least a "best attempt" can be made; if restore is possible (model is unchanged, foreign keys are unchanged, etc.) - great; if not - en error message can be generated informing that it is not possible to restore the object (and give an explanation).
